geonode/security/utils.py

```diff
@@ -324,21 +274,348 @@ def get_user_visible_groups(user, include_public_invite: bool = False):
+    def is_advanced_workflow():
+        """
+          **ADVANCED WORKFLOW**
+            - `RESOURCE_PUBLISHING = True`
+            - `ADMIN_MODERATE_UPLOADS = True`
+
+            - When user creates a resource:
+              - OWNER gets all the owner permissions (`publish_resource` and `change_resourcebase_permissions` EXCLUDED)
+              - Group MANAGERS of the user's groups will get the owner permissions (`publish_resource` INCLUDED)
+              - Group MEMBERS of the user's groups will get the `view_resourcebase`, `download_resourcebase` permission
+              - ANONYMOUS can not view and download if the resource is not published
+
+            - When resource has a group assigned:
+              - OWNER gets all the owner permissions (`publish_resource` and `change_resourcebase_permissions` EXCLUDED)
+              - Group MANAGERS of the resource's group will get the owner permissions (`publish_resource` INCLUDED)
+              - Group MEMBERS of the resource's group will get the `view_resourcebase`, `download_resourcebase` permission
+        """
+        return settings.RESOURCE_PUBLISHING and settings.ADMIN_MODERATE_UPLOADS
+
+    @staticmethod
+    def is_simplified_workflow():
+        """
+          **SIMPLIFIED WORKFLOW**
+            - `RESOURCE_PUBLISHING = False`
+            - `ADMIN_MODERATE_UPLOADS = True`
+
+            - **NOTE**: Is it even possibile? when the resource is automatically published, can it be un-published?
+            If this combination is not allowed, we should either stop the process when reading the settings or log a warning and force a safe combination.
+
+            - When user creates a resource:
+              - OWNER gets all the owner permissions (`publish_resource` and `change_resourcebase_permissions` INCLUDED)
+              - Group MANAGERS of the user's groups will get the owner permissions (`publish_resource` INCLUDED)
+              - Group MEMBERS of the user's group will get the `view_resourcebase`, `download_resourcebase` permission
+              - ANONYMOUS can view and download
+        """
+        return not settings.RESOURCE_PUBLISHING and settings.ADMIN_MODERATE_UPLOADS
+
+    @staticmethod
+    def assignable_perm_condition(perm, resource_type):
+        _assignable_perm_policy_condition = (perm in DOWNLOAD_PERMISSIONS and resource_type in DOWNLOADABLE_RESOURCES) or \
+            (perm in DATASET_EDIT_DATA_PERMISSIONS and resource_type in DATA_EDITABLE_RESOURCES_SUBTYPES) or \
+            (perm not in (DOWNLOAD_PERMISSIONS + DATASET_EDIT_DATA_PERMISSIONS))
+        logger.debug(f" perm: {perm} - resource_type: {resource_type} --> assignable: {_assignable_perm_policy_condition}")
+        return _assignable_perm_policy_condition
+
+    @staticmethod
+    def get_instance(uuid: str):
+        from geonode.base.models import ResourceBase
+
+        _resources = ResourceBase.objects.filter(uuid=uuid)
+        _exists = _resources.count() == 1
+        if _exists:
```

Comment:
This can be written also in this way:

 ```python
from geonode.base.models import ResourceBase
return ResourceBase.objects.filter(uuid=uuid).first()
```

is almost impossible that we can have two resources with one UUID and if we want to force that, maybe is better to change the model and make it unique

Reply:
done